provider/ansible/example.rb

```diff
@@ -92,6 +92,70 @@ def build_task(state, object)
 
       private
 
+      def verbs
+        {
+          present: 'created',
+          absent: 'deleted',
+          facts: 'verify'
+        }
+      end
+    end
+
+    # Holds all information necessary to run a facts module and verify the
+    # creation / deletion of a resource.
+    #
+    # command         - The gcloud command being run
+    # failed_name     - The name of the resource being created
+    #                   (defaults to {{resource_name}})
+    #                   Used for verifying deletion.
+    # failed_verifier - A custom Jinja2 verifier to test if gcloud
+    #                   command post-deletion worked as intended.
+    class FactsVerifier < Api::Object
+      include Compile::Core
+
+      def validate
+        @failure ||= FailureCondition.new
+
+        check_property :parameters, Hash
+      end
+
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/MethodLength
+      def build_task(state, object)
+        obj_name = Google::StringUtils.underscore(object.name)
+        verb = verbs[state.to_sym]
+        number = state == 'present' ? 1 : 0
+        module_name = ["gcp_#{object.__product.prefix[1..-1]}",
+                       Google::StringUtils.underscore(object.name),
+                       'facts'].join('_')
+        [
+          "- name: verify that #{obj_name} was #{verb}",
```

Comment:
There are a lot of formatting things in here- would it make sense for this part to be an erb template? Or is there enough regular ruby going on that this is better?

Reply:
For the moment, I think it's alright to keep this as Ruby. I'm going to do a small refactor of this code in a separate PR. It should clean it up and make generating facts examples from regular examples possible.

Reply:
As part of #343, I ended up using ERB templates. Thanks for the idea!

#343 should also enable me to build the facts examples from the regular examples, which would be super awesome.